animal_crossing.py

```diff
@@ -0,0 +1,187 @@
+		''')
+		return
+
+	split = cmd.args.split(' ', 1)
+	subcmd = split[0]
+	subargs = len(split) == 2 and split[1] or ''
+	if subcmd == 'tz':
+		_stalk_set_timezone(cmd, subargs)
+	elif subcmd == 'sell':
+		_stalk_set_sell_price(cmd, subargs)
+	elif subcmd == 'trigger':
+		_stalks_set_sell_trigger(cmd, subargs)
+	else:
+		cmd.reply('Unrecognized stalks subcommand %s' % subcmd)
+
+def _stalk_set_sell_price(cmd, price):
+	if not price:
+		_stalk_list_sale_prices(cmd)
+		return
+
+	user_id = cmd.sender['id']
+	current_time = datetime.now(timezone.utc)
+	cur = db.execute('SELECT timezone FROM user WHERE id=?', (user_id,))
+	res = cur.fetchone()
+	if not res:
+		cmd.reply('Could not add sale price. Have you registered a friend code?')
+		return
+	elif res['timezone'] is None:
+		cmd.reply('Could not add sale price. Please register a time zone with !stalks tz')
+		return
+
+	user_time = current_time.astimezone(dateutil.tz.gettz(res['timezone']))
+	if (user_time.hour < 8 or user_time.hour >= 22):
+		cmd.reply('Your shops are closed. Your current time zone is %s, where it is currently %s.' %
+			(res['timezone'], user_time.strftime(time_format)))
+		return
+	elif user_time.weekday() == 6:
+		cmd.reply('It is currently Sunday in your selected time zone, %s. Turnip offers cannot be submitted.' %
+			(res['timezone']))
+		return
+
+	if user_time.hour >= 12:
+		expiration = user_time.replace(hour=22, minute=0, second=0, microsecond=0)
+	else:
+		expiration = user_time.replace(hour=12, minute=0, second=0, microsecond=0)
+
+	expires_in = _calculate_expiration(expiration - user_time)
+
+	try:
+		value = int(price)
+	except ValueError:
+		cmd.reply('Could not parse sell value. Usage: !stalks sell 123')
+		return
+
+	try:
+		with db:
+			db.execute('INSERT INTO sell_price VALUES (?, ?, ?, ?)',
+				(user_id, current_time, expiration.astimezone(timezone.utc), value))
+		cmd.reply('Sale price recorded at %s bells. Offer expires in %dh %dm.' %
+			(price, expires_in[0], expires_in[1]))
+		_stallk_check_sell_triggers(cmd, price, expires_in)
+	except sqlite3.IntegrityError:
+		cmd.reply('Could not add sale price. Have you registered a friend code?')
+
+def _stallk_check_sell_triggers(cmd, price, expires_in):
+	cur = db.execute('''
+	SELECT user_id FROM sell_trigger WHERE sell_trigger.price <= ?
+	''', (price,))
+
+	triggers = [x['user_id'] for x in cur.fetchall() if x['user_id'] != cmd.sender['id']]
+	if triggers:
+		msg = ' '.join(['<@!%s>' % (x) for x in triggers])
+		msg += (': %s has reported a sell price of %s, above your configured trigger. Their offer will expire in %dh%dm.' %
+			(cmd.sender['username'], price, expires_in[0], expires_in[1]))
+		cmd.reply(msg)
+
+def _stalk_list_sale_prices(cmd):
+	global cached_row_id
+	cur = db.execute('''
+	SELECT sell_price.rowid, *
+	FROM sell_price
+	INNER JOIN user ON sell_price.user_id = user.id
+	WHERE sell_price.rowid > ?
+	ORDER BY sell_price.rowid ASC
+	''', (cached_row_id,))
```

Comment:
ISO-formatted datetimes are lexicographically ordered chronologically so you can just `WHERE expiration > NOW()` or something instead of using `cached_row_id`